ibis/expr/api.py

```diff
@@ -4193,22 +4193,39 @@ def mutate(table, exprs=None, **mutations):
         for name, expr in sorted(mutations.items(), key=operator.itemgetter(0))
     )
 
+    overwriting_cols_to_expr = {}
```

Comment:
wow this is extra complicated :->

can you add some comments for a future reader

Reply:
Yep, added, hopefully a bit more clear now